src/content/it/fundamentals/performance/rendering/optimize-javascript-execution.md

```diff
@@ -0,0 +1,192 @@
+causing the browser to miss a frame.">
+
+Infatti, jQuery utilizzava `setTimeout` per il suo comportamento `animate` . È
+stato modificato per utilizzare `requestAnimationFrame` nella versione 3. Se
+utilizzi la versione precedente di jQuery, è possibile <a
+href="https://github.com/gnarf/jquery-requestAnimationFrame"
+data-md-type="link">applicare una patch per utilizzare
+`requestAnimationFrame`</a>, come fortemente consigliato.
+
+## Riduci la complessità o utilizza Web Workers
+
+JavaScript gira sul thread principale del browser, accanto a calcoli di stile,
+layout e, in molti casi, paint. Se il tuo JavaScript viene eseguito per un lungo
+periodo, bloccherà queste altre attività, causando potenzialmente la perdita dei
+frame.
+
+Dovresti essere tattico su quando eseguire JavaScript e per quanto tempo. Ad
+esempio, se ti trovi in ​​un'animazione come lo scorrimento, dovresti idealmente
+cercare di mantenere il tuo JavaScript pressapoco per **3-4 ms** . Esecuzioni
+prolungate rischiano di perdere troppo tempo. Se sei in un periodo di idle, puoi
+permetterti di essere più rilassato riguardo al tempo impiegato.
+
+In molti casi è possibile spostare il lavoro di calcolo puro sui [Web
+Worker](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/basic_usage)
+se, ad esempio, non richiedi accesso DOM. La manipolazione o l'attraversamento
+dei dati, come l'ordinamento o la ricerca, sono spesso adatte a questo modello,
+così come il caricamento e la generazione del modello.
+
+```
+var dataSortWorker = new Worker("sort-worker.js");
+dataSortWorker.postMesssage(dataToSort);
+
+// The main thread is now free to continue working on other things...
+
+dataSortWorker.addEventListener('message', function(evt) {
+   var sortedData = evt.data;
+   // Update data on screen...
+});
+```
+
+Non tutto il lavoro può adattarsi a questo modello: i Web Workers non hanno
+accesso DOM. Laddove il tuo lavoro deve essere sul thread principale, considera
+un approccio di batching, in cui segmenti una grande attività in micro-task,
+ognuno dei quali richiede non più di qualche millisecondo, da eseguire
+all'interno dei gestori `requestAnimationFrame` su ogni frame.
+
+```
+var taskList = breakBigTaskIntoMicroTasks(monsterTaskList);
+requestAnimationFrame(processTaskList);
+
+function processTaskList(taskStartTime) {
+  var taskFinishTime;
+
+  do {
+    // Assume the next task is pushed onto a stack.
+    var nextTask = taskList.pop();
+
+    // Process nextTask.
+    processTask(nextTask);
+
+    // Go again if there’s enough time to do the next task.
+    taskFinishTime = window.performance.now();
+  } while (taskFinishTime - taskStartTime < 3);
+
+  if (taskList.length > 0)
+    requestAnimationFrame(processTaskList);
+
+}
+```
+
+Ci sono conseguenze UX e UI per questo approccio e dovrai assicurarti che
+l'utente sappia che un compito è in fase di elaborazione, [usando un indicatore
```

Comment:
l'utente sappia quando un'attività è in fase di elaborazione, [usando un indicatore